### PR TITLE
CoC- Ikoma Tsanuri

### LINKED ISSUE #366: CoC - Ikoma Tsanuri
image

### PR DESCRIPTION
**Pull request by marcsoiferman**
Closes #366

### CODE REVIEW

File: test/server/cards/14.4-CoC/IkomaTsanuri2.spec.js [hunk 1]
```diff
@@ -0,0 +1,170 @@
+describe('Countryside Trader', function() {
```
Comment (The-Outcast) [inline test/server/cards/14.4-CoC/IkomaTsanuri2.spec.js:1]: Bit too fast on the copy pasta 😆

File: server/game/basecard.ts [hunk 2]
```diff
@@ -364,8 +364,8 @@ class BaseCard extends EffectSource {
         }
     }
 
-    canTriggerAbilities(context: AbilityContext): boolean {
-        return !this.facedown && this.checkRestrictions('triggerAbilities', context);
+    canTriggerAbilities(context: AbilityContext, ignoredRequirements = []): boolean {
+        return !this.facedown && (ignoredRequirements.includes('triggeringRestrictions') || this.checkRestrictions('triggerAbilities', context));
```
Comment (The-Outcast) [inline server/game/basecard.ts:368]: Looking at the merge that needs to happen `|| this.checkRestrictions('triggerAbilities', context))` actually is a && case. Any reason why this changed into OR here?

Comment (marcsoiferman) [reply server/game/basecard.ts:368]: My best guess is that I changed it from
!facedown AND [no restrictions on triggering abilities]
to
!facedown AND [either (I ignoring triggering restrictions) or (there are no triggering restrictions)]

Comment (marcsoiferman) [reply server/game/basecard.ts:368]: Ah yes, looked at it more closely.  We used the same pathways to do "Trigger the ability" vs "Resolve the ability" (ex: Cunning Negotiator vs Countryside Trader).  So I had to put a "ignore triggering restriction" when we do direct "resolve the ability".

**Pull request merged**